src/core/Rule.ml

```diff
@@ -173,6 +173,7 @@ and metavar_cond =
    *)
   | CondRegexp of
       MV.mvar * Xpattern.regexp_string * bool (* constant-propagation *)
+  | CondType of MV.mvar * Xlang.t option * string * AST_generic.type_
```

Comment:
can you add a comment about what is the string and why we need an Xlang.t here?

Reply:
Maybe also add a comment like (* LATER: we could also parse the type lazily, like we for the patterns *)